### PR TITLE
Fix db query is not get when using scalikejdbc

### PR DESCRIPTION
**Pull request by orekyuu**
refs: #1006

## Before
ResourceName always `DB Query` when using scalikejdbc.

## After
Get query string even if when use scalikejdbc.

### CODE REVIEW

**Comment by richardstartin** [comment]
Thanks for the contribution, it looks good except for one complication. The test on JDK7 is failing because `scalike` is compiled to JDK8 bytecode. We have JDK7 users and I think JDBC is probably far too common to baseline at JDK8. One solution would be to create a `scalike` instrumentation baselined at JDK8, and to extend the JDBC decorator in that instrumentation.

File: dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/PreparedStatementInstrumentation.java [hunk 1]
```diff
@@ -39,7 +40,8 @@ public PreparedStatementInstrumentation() {
 
   @Override
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return implementsInterface(named("java.sql.PreparedStatement"));
+    return implementsInterface(named("java.sql.PreparedStatement"))
+        .and(not(named("scalikejdbc.DBConnectionAttributesWiredPreparedStatement")));
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/PreparedStatementInstrumentation.java:44]: @orekyuu This seems to be the only functional change. Can you better explain how this solves the problem?

Comment (orekyuu) [reply dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/PreparedStatementInstrumentation.java:44]: Before the change, the following was executed.

1. Call prepareStatement method in `scalikejdbc.DBSession#createStatementExecutor`. (https://github.com/scalikejdbc/scalikejdbc/blob/3f24bdb57af36eeb2ea636710e8845773fad19f1/scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala#L87 )
1. In `JDBCMaps.preparedStatements`, save SQL with raw PreparedStatement as key.
1. DBSession wraps PreparedStatement with DBConnectionAttributesWiredPreparedStatement ( https://github.com/scalikejdbc/scalikejdbc/blob/3f24bdb57af36eeb2ea636710e8845773fad19f1/scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala#L93 )
1. Get SQL using JDBCConnectionAttributesWiredPreparedStatement from JDBCMaps.preparedStatements but no value exists.

Ignoring the DBConnectionAttributesWiredPreparedStatement and using the raw PreparedStatement should work as expected. 🤔

Comment (tylerbenson) [reply dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/PreparedStatementInstrumentation.java:44]: Ok, so ignoring the wrapped one allows the instrumentation to be properly applied on the "underlying" one?

Comment (orekyuu) [reply dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/PreparedStatementInstrumentation.java:44]: When JDBCConnectionAttributesWiredPreparedStatement#execute was called, span was recorded there. However, if JDBCConnectionAttributesWiredPreparedStatement is ignored, the span will be recorded by the execute of another PreparedStatement (for example, HikariProxyPreparedStatement), so it looks like the expected behavior.

**Comment by tylerbenson** [comment]
It seems the only thing that is Java 8 specific is the test, so perhaps it could be moved to a separate `scalalikejdbc` project that only contains the tests?

**Comment by orekyuu** [comment]
Looks like unrelated tests are failing.
Is this a random fail?

**Comment by richardstartin** [comment]
Yes just flaky tests, unrelated to this change. I've resubmitted them to get a clean build.

File: settings.gradle [hunk 2]
```diff
@@ -145,6 +145,7 @@ include ':dd-java-agent:instrumentation:reactor-core-3.1'
 include ':dd-java-agent:instrumentation:rmi'
 include ':dd-java-agent:instrumentation:rxjava-1'
 include ':dd-java-agent:instrumentation:scala-concurrent'
+include ':dd-java-agent:instrumentation:scalikejdbc'
```
Comment (tylerbenson) [inline settings.gradle:148]: One last request... please move this to `:dd-java-agent:instrumentation:jdbc:scalikejdbc` then we'll be good to go.

**Pull request merged**